src/app.controller.ts

```diff
@@ -15,16 +15,12 @@ export class AppController {
   getBoardStateFromFile(
     @Query('x') x?: number,
     @Query('y') y?: number,
-  ): { gameState: GameState } {
+  ): { gameState: GameState; winner: StoneColor } {
     const gameState = this.appService.getBoardStateFromFile();
-    return x && y
-      ? {
-          gameState: this.appService.updateGameState(gameState, {
-            x,
-            y,
-          }),
-        }
-      : { gameState };
+    const nextGameState =
+      x && y ? this.appService.updateGameState(gameState, { x, y }) : gameState;
+    const hasAWinner = getWinner(nextGameState);
```

Comment:
In the same way, I feel like the win detection could be part of the `updateGameState()` function (inside a sub-function, of course)